gitlab-runner.yaml

```diff
@@ -288,13 +292,11 @@ Resources:
                       sudo gitlab-runner register \
                       --non-interactive \
                       --url "${GitlabServerURL}" \
-                      --registration-token $GitlabGroupToken \
+                      --token $GitlabGroupToken \
                       --executor "docker" \
                       --docker-image "${DockerImagePath}" \
                       --description "Gitlab Runner with Docker Executor" \
-                      --locked="${isLOCKED}" --access-level "${ACCESS}" \
-                      --docker-volumes "/var/run/docker.sock:/var/run/docker.sock" \
-                      --tag-list "${RunnerEnvironment}-${RunnerVersion}-docker"
+                      --docker-volumes "/var/run/docker.sock:/var/run/docker.sock" 
                   done
                 - isLOCKED: !FindInMap [GitlabRunnerRegisterOptionsMap, !Ref RunnerEnvironment, isLOCKED]
```

Comment:
Should these lines: isLOCKED and ACCESS also be removed since we are removing the --locked and --access-level options?

Reply:
Based on this [reference](https://docs.gitlab.com/runner/register/#legacy-compatible-registration-process)
> Runner registration tokens and several runner configuration arguments were [deprecated](https://gitlab.com/gitlab-org/gitlab/-/issues/379743) in GitLab 15.6 and will be removed in GitLab 18.0. To ensure minimal disruption to your automation workflow, the legacy-compatible registration process triggers if a runner authentication token is specified in the legacy parameter --registration-token.
> 
> The legacy-compatible registration process ignores the following command-line parameters. These parameters can only be configured when a runner is created in the UI or with the API.
> 
> --locked
> --access-level
> --run-untagged
> --maximum-timeout
> --paused
> --tag-list
> --maintenance-note

I succeeded in testing this configuration.

Reply:
Got it. What I meant was should we also remove these lines (lines 301 and 302 in your version):
- isLOCKED: !FindInMap [GitlabRunnerRegisterOptionsMap, !Ref RunnerEnvironment, isLOCKED]
  ACCESS: !FindInMap [GitlabRunnerRegisterOptionsMap, !Ref RunnerEnvironment, ACCESS]